core/Pos/Core/Timestamp.hs

```diff
@@ -59,6 +61,9 @@ diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
 addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
 addMicrosecondsToTimestamp m t = Timestamp { getTimestamp = (getTimestamp t) + m }
 
+timestampToPosix :: Timestamp -> POSIXTime
+timestampToPosix (Timestamp ts) = fromIntegral ts / 1000000
```

Comment:
You could add this to the `TimeUnit` typeclass. Ideally, the signature would be something like `toUnitInteger :: (Coercible b Integer, TimeUnit b) => a -> b -> Integer`, but you can make it more specific with the default implementation.